cmd/workspace/groups/groups.go

```diff
@@ -292,31 +274,31 @@ var patchCmd = &cobra.Command{
 	RunE: func(cmd *cobra.Command, args []string) (err error) {
 		ctx := cmd.Context()
 		w := root.WorkspaceClient(ctx)
+
 		if cmd.Flags().Changed("json") {
 			err = patchJson.Unmarshal(&patchReq)
 			if err != nil {
 				return err
 			}
-		} else {
-			if len(args) == 0 {
-				promptSpinner := cmdio.Spinner(ctx)
-				promptSpinner <- "No ID argument specified. Loading names for Groups drop-down."
-				names, err := w.Groups.GroupDisplayNameToIdMap(ctx, iam.ListGroupsRequest{})
-				close(promptSpinner)
-				if err != nil {
-					return fmt.Errorf("failed to load names for Groups drop-down. Please manually specify required arguments. Original error: %w", err)
-				}
-				id, err := cmdio.Select(ctx, names, "Unique ID for a group in the Databricks workspace")
-				if err != nil {
-					return err
-				}
-				args = append(args, id)
+		}
+		if len(args) == 0 {
+			promptSpinner := cmdio.Spinner(ctx)
+			promptSpinner <- "No ID argument specified. Loading names for Groups drop-down."
+			names, err := w.Groups.GroupDisplayNameToIdMap(ctx, iam.ListGroupsRequest{})
+			close(promptSpinner)
+			if err != nil {
+				return fmt.Errorf("failed to load names for Groups drop-down. Please manually specify required arguments. Original error: %w", err)
 			}
-			if len(args) != 1 {
-				return fmt.Errorf("expected to have unique id for a group in the databricks workspace")
+			id, err := cmdio.Select(ctx, names, "Unique ID for a group in the Databricks workspace")
+			if err != nil {
+				return err
 			}
-			patchReq.Id = args[0]
+			args = append(args, id)
+		}
+		if len(args) != 1 {
```

Comment:
This will break I believe?

Reply:
@mgyucht no, why? This is a general logic for prompts. If no args provided it will go to the condition on line 284 and ask user to choose group ID from the list and append it to args on line 296 so `len(args) == 1`

Or if customer provided positional argument, the prompt condition will be skipped

Reply:
My mistake, didn't realize args itself was modified in that block.